pkg/mountinfo/mountinfo_linux.go

```diff
@@ -0,0 +1,117 @@
+// +build linux
+
+/*
+ * Minio Cloud Storage, (C) 2017 Minio, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package mountinfo
+
+import (
+	"bufio"
+	"fmt"
+	"io"
+	"os"
+	"path/filepath"
+	"strconv"
+	"strings"
+)
+
+const (
+	// Number of fields per line in /proc/mounts as per the fstab man page.
+	expectedNumFieldsPerLine = 6
+	// Location of the mount file to use
+	procMountsPath = "/proc/mounts"
+)
+
+// List - lists all mounts.
+func List() (Mounts, error) {
+	return readProcMounts(procMountsPath)
+}
+
+// CheckCrossDevice - check if given path has any sub-mounts in the input mounts list.
+func (mts Mounts) CheckCrossDevice(path string) error {
```

Comment:
if the signature of this function is:

```
CheckCrossDevice(paths []string) error
```

We can avoid exposing Mount (as this struct is not useful outside the package)

i.e the package can expose only one function